### DISCUSSION TITLE
Disabling "Press Alt + E to view envelope mode in piano roll" for Live 12

### DISCUSSION THREAD

**Discussion by MaximillionBowie** [LiveEnhancementSuite/LESforMacOS#42] [Q&A]
Hi All!
I am happy to report that LES is still working like a wizard in Live 12 Beta on my Macbook Pro M1 Max running Ventura )

I'm not sure if this would be considered an issue or a discussion, so I'm posting it in both :

There is one exception - Live 12 has a new, long-sought-after editing command to freely chop midi notes in piano scroll by pressing "alt-e" and then click-dragging to the desired number of equally spaced notes. It's pretty brilliant ;)

The rub is, this conflicts with LES's current "Press Alt + E to view envelope mode in piano roll" function, which isn't needed anymore with Live 12's new Envelope mode hot key, and creates cursor havoc when trying to use the midi editing function I mentioned earlier.

Can the "Press Alt + E to view envelope mode in piano roll" function be disabled in LES for use with Live 12? Perhaps there's a way to modify LES to include  a "1 or 0" toggle option for it  in "Configure Settings"?

Any experts who could point me in the right direction?

LES with Ableton is still the best daw going - keep rockin n rollin ! ! !

**Comment by ka-maji** [comment]
Just shared this fix I found on the Ableton subreddit, will paste below:

I've found a fix! It's conflicting with a LES feature called 'envelope mode macro'. If you disable that (I don't use it), then the alt+e shortcut works again. I'll give instructions for the fix below:

- Assuming you have LES installed in your applications folder, go to _Applications/Live Enhancement Suite/Contents/Resources/extensions/hs/les/LESmain.lua_ (right click and show package contents on Live Enhancement Suite to get there)
- I'd advise making a copy of _LESmain.lua_ and save it elsewhere, because we're going to edit it
- Open _LESmain.lua_ in TextEdit. Hit command+f, then search for 'envelope mode'. There should be a section of script for the envelope mode, which uses the alt+e shortcut.
- Select everything from `--envelope mode macro` all the way up to the `end` line that occurs just before `-- save as new version`. Delete all that text.
- Save LESmain.lua, quit LES and restart. Hopefully this feature will work now! Got it working on my machine.

**Reply by MaximillionBowie** [reply]
@ka-maji 

It Worked! Thanks a million for this! WooHOOOOOOOO ! ! !
